### PR TITLE
TimePicker: Fix issue with previous fiscal quarter not parsing correctly

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- fixes a bug where using the `Previous fiscal quarter` option with a custom fiscal year start month would not parse correctly
- adds some unit tests to prevent regressions

**Why do we need this feature?**

- when setting a fiscal year start month to later in the year the datemath parsing would not work correctly

**Who is this feature for?**

- anyone using previous fiscal quarters

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/support-escalations/issues/6609

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-data/src/datetime/datemath.ts [hunk 1]
```diff
@@ -163,15 +163,11 @@ export function parseDateMath(
...
+        if (isFiscal) {
+          roundToFiscal(fiscalYearStartMonth, dateTime, unit, roundUp);
         } else {
-          if (isFiscal) {
-            roundToFiscal(fiscalYearStartMonth, dateTime, unit, roundUp);
+          if (roundUp) {
+            dateTime.endOf(unit);
           } else {
             dateTime.startOf(unit);
           }
```
Comment (ashharrison90) [inline packages/grafana-data/src/datetime/datemath.ts:173]: this is a small bit of refactoring which isn't actually relevant to the fix

File: packages/grafana-data/src/datetime/datemath.ts [hunk 2]
```diff
@@ -199,7 +195,8 @@ export function roundToFiscal(fyStartMonth: number, dateTime: any, unit: string,
       if (roundUp) {
         roundToFiscal(fyStartMonth, dateTime, unit, false).add(2, 'M').endOf('M');
       } else {
-        dateTime.subtract((dateTime.month() - fyStartMonth + 3) % 3, 'M').startOf('M');
+        // why + 12? to ensure this number is always a positive offset from fyStartMonth
+        dateTime.subtract((dateTime.month() - fyStartMonth + 12) % 3, 'M').startOf('M');
```
Comment (ashharrison90) [inline packages/grafana-data/src/datetime/datemath.ts:199]: this + 12 is the key 🔑

File: packages/grafana-data/src/datetime/datemath.ts (hunk 1 above)
Comment (JoaoSilvaGrafana) [reply packages/grafana-data/src/datetime/datemath.ts:173]: It does look clearer though 👏

**Review by JoaoSilvaGrafana — APPROVED** [review]
Lgtm! 🎉

**Pull request merged**